config/config/src/index.ts

```diff
@@ -516,6 +517,10 @@ export async function getConfig (opts: {
     pnpmConfig.dev = true
   }
 
+  if (pnpmConfig.dangerouslyAllowAllBuilds) {
+    pnpmConfig.neverBuiltDependencies = []
```

Comment:
If `pnpmConfig.neverBuiltDependencies` has a value, we should probably give a hint message.